Packs/CIRCL/Integrations/CirclCVESearch/CirclCVESearch.py

```diff
@@ -0,0 +1,301 @@
+
+
+def parse_cpe(cpes: list[str], cve_id: str) -> tuple[list[str], list[EntityRelationship]]:
+    """
+    Parses a CPE to return the correct tags and relationships needed for the CVE.
+
+    Args:
+        cpe: A list representing a single CPE, see "https://nvlpubs.nist.gov/nistpubs/legacy/ir/nistir7695.pdf"
+
+    Returns:
+        A tuple consisting of a list of tags and a list of EntityRelationships.
+
+    """
+
+    cpe_parts = {
+        "a": "Application",
+        "o": "Operating-System",
+        "h": "Hardware"
+    }
+
+    vendors = set()
+    products = set()
+    parts = set()
+
+    for cpe in cpes:
+        cpe_split = re.split('(?<!\\\):', cpe)
+
+        try:
+            vendor = cpe_split[3].capitalize().replace("\\", "").replace("_", " ")
+            if vendor:
+                vendors.add(vendor)
+
+        except IndexError:
+            pass
+
+        try:
+            product = cpe_split[4].capitalize().replace("\\", "").replace("_", " ")
+            if product:
+                products.add(product)
+
+        except IndexError:
+            pass
+
+        try:
+            parts.add(cpe_parts[cpe_split[2]])
+
+        except IndexError:
+            pass
+
+    relationships = [EntityRelationship(name="targets",
+                                        entity_a=cve_id,
+                                        entity_a_type="cve",
+                                        entity_b=vendor,
+                                        entity_b_type="identity") for vendor in vendors]
+
+    relationships.extend([EntityRelationship(name="targets",
+                                             entity_a=cve_id,
+                                             entity_a_type="cve",
+                                             entity_b=product,
+                                             entity_b_type="software") for product in products])
+
+    return list(vendors | products | parts), relationships
+
+
+def generate_indicator(data: dict) -> Common.CVE:
+    """
+    Generating a single cve indicator with dbot score from cve data.
+
+    Args:
+        data: The cve data
+
+    Returns:
+        A CVE indicator with dbotScore
+    """
+
+    cve_id = data.get('id', '')
+    cpe = data.get("vulnerable_product", '')
+
+    if cpe:
+        tags, relationships = parse_cpe(cpe, cve_id)
+
+    else:
+        relationships = []
+        tags = []
+
+    cwe = data.get('cwe', '')
+
+    if cwe and cwe != 'NVD-CWE-noinfo':
+        tags.append(cwe)
+
+    cvss_table = []
+
+    for category in ("impact", "access"):
+        for key, value in data.get(category, []).items():
+            cvss_table.append({"metrics": key, "value": value})
+
+    vulnerable_products = [Common.CPE(cpe) for cpe in data.get("vulnerable_product", [])]
+    vulnerable_configurations = [Common.CPE(cpe["id"]) for cpe in data.get("vulnerable_configuration", [])]
+    cpes = set(vulnerable_products) | set(vulnerable_configurations)
+
+    cve_object = Common.CVE(
```

Comment:
What about DBOT score?

Reply:
CVE doesnt really need one due to the reputation script it uses which assigns a score according to cvss